test/contconic.jl

```diff
@@ -506,6 +506,70 @@ function contconictest(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64))
         end
     end
 
+    if MOI.supportsproblem(solver, MOI.ScalarAffineFunction{Float64}, [(MOI.ScalarAffineFunction{Float64},MOI.Zeros),(MOI.VectorVariablewiseFunction,MOI.SecondOrderCone)])
+        @testset "SOC4" begin
+            # Problem SOC4
+            # min 0x[1] - 2x[2] - 1x[3]
+            #  st  x[1]                                == 1 (c1a)
+            #              x[2]         - x[4]         == 0 (c1b)
+            #                      x[3]         - x[5] == 0 (c1c)
+            #      x[1] >= ||(x[4],x[5])||                  (c2)
+            # in conic form:
+            # min  c^Tx
+            # s.t. Ax + b ∈ {0}₃
+            #      (x[1],x[4],x[5]) ∈ SOC₃
+            # Like SOCINT1 but with copies of variables and integrality relaxed
+            # Tests out-of-order indices in cones
+
+            b = [1.0,0.0,0.0]
+            A = [1.0 0.0 0.0 0.0 0.0
+                0.0 1.0 0.0 -1.0 0.0
+                0.0 0.0 1.0 0.0 -1.0]
+            c = [0.0,-2.0,-1.0,0.0,0.0]
+
+            m = MOI.SolverInstance(solver)
+
+            x = MOI.addvariables!(m, 5)
+            
+            A_cols = x
+            A_rows = [1,2,3,2,3]
+            A_vals = [1.0,1.0,1.0,-1.0,-1.0]
+
+            c1 = MOI.addconstraint!(m, MOI.VectorAffineFunction(A_rows,A_cols,A_vals,b), MOI.Zeros(3))
+            c2 = MOI.addconstraint!(m, MOI.VectorVariablewiseFunction([x[1],x[4],x[5]]), MOI.SecondOrderCone(3))
+
+            @test MOI.getattribute(m, MOI.NumberOfConstraints{MOI.VectorAffineFunction{Float64},MOI.Zeros}()) == 1
+            @test MOI.getattribute(m, MOI.NumberOfConstraints{MOI.VectorVariablewiseFunction,MOI.SecondOrderCone}()) == 1
+
+            MOI.setobjective!(m, MOI.MinSense, MOI.ScalarAffineFunction(x,c,0.0))
+            MOI.optimize!(m)
+
+            @test MOI.cangetattribute(m, MOI.TerminationStatus())
+            @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.Success
+
+            @test MOI.cangetattribute(m, MOI.PrimalStatus())
+            @test MOI.getattribute(m, MOI.PrimalStatus()) == MOI.FeasiblePoint
+            @test MOI.cangetattribute(m, MOI.DualStatus())
+            @test MOI.getattribute(m, MOI.DualStatus()) == MOI.FeasiblePoint
+
+            @test MOI.cangetattribute(m, MOI.VariablePrimal(), x)
+            x_primal = MOI.getattribute(m, MOI.VariablePrimal(), x)
+            @test x_primal[1]^2 ≥ x_primal[4]^2 + x_primal[5]^2 - ε
+
+            @test MOI.cangetattribute(m, MOI.VariableDual(), x)
```

Comment:
`VariableDual` is not a thing anymore. It should be `ConstraintDual` on the reference `c2`.